protoarray/src/main/java/tech/pegasys/artemis/protoarray/ProtoArrayForkChoiceStrategy.java

```diff
@@ -139,13 +139,15 @@ private static void processBlockAtStartup(
         store.getBlockState(block.hash_tree_root()).getFinalized_checkpoint().getEpoch());
   }
 
-  void processAttestation(int validatorIndex, Bytes32 blockRoot, UnsignedLong targetEpoch) {
-    VoteTracker vote = votes.get(validatorIndex);
+  void processAttestation(MutableStore store, int validatorIndex, Bytes32 blockRoot, UnsignedLong targetEpoch) {
+    VoteTracker vote = store.getVote(validatorIndex).copy();
 
     if (targetEpoch.compareTo(vote.getNextEpoch()) > 0 || vote.equals(VoteTracker.Default())) {
       vote.setNextRoot(blockRoot);
       vote.setNextEpoch(targetEpoch);
     }
+
+    store.setVote(validatorIndex, vote);
```

Comment:
If you move `getVote` into `Transaction` then you won't need to do this - just get the mutable vote and edit it as needed.  When the transaction commits those changes can all be committed as well.

Reply:
Makes sense.